gematria/datasets/python/import_annotated_basic_blocks.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Creates an annotated Gematria data set from an ELF object and perf samples.
+
+Reads basic blocks from an ELF object, along with event samples and block
+latencies from a `perf.data`-like file, usually generated using `perf record`,
+and writes a Gematria TFRecord dataset containing basic blocks with instruction
+annotations as derived from the samples.
+"""
+
+from collections.abc import Sequence
+
+from absl import app
+from absl import flags
+from absl import logging
+from gematria.datasets.python import annotating_importer
+from gematria.llvm.python import canonicalizer
+from gematria.llvm.python import llvm_architecture_support
+from gematria.proto import basic_block_pb2
+from pybind11_abseil import status
+import tensorflow as tf
+
+
+_INPUT_ELF_FILE = flags.DEFINE_string(
+    'gematria_input_elf',
+    None,
+    'The name of the ELF file from which basic blocks are to be imported',
+    required=True,
+)
+_INPUT_PERF_FILE = flags.DEFINE_string(
+    'gematria_input_perf',
```

Comment:
Very nit: `gematria_input_perf_data` for best matching between the file name and the flag.